### PR TITLE
virtio-fs: new case to check file's stdev

### PR DESCRIPTION
**Pull request by xiagao**
Get file's st_dev of different device.
ID: 1978153
Signed-off-by: Xiaoling Gao <xiagao@redhat.com>

### CODE REVIEW

**Comment by xiagao** [comment]
@YongxueHong Could you also review this patch ? Thanks.

**Comment by YongxueHong** [comment]
CC @zhenyzha 
Please help check it on ppc, thanks.

File: qemu/tests/virtio_fs_share_data.py [hunk 1]
```diff
@@ -296,6 +306,32 @@ def get_viofs_exe(session):
...
+            if cmd_get_stdev:
+                def get_stdev(file):
+                    """
+                    Get file's st_dev value.
+                    """
+                    stdev = session.cmd_output(cmd_get_stdev % file).strip()
+                    pattern = r'Device:.*/(\w+)\s'
+                    st_dev_id = re.findall(pattern, stdev, re.I)
+                    logging.info("%s device id is %s.", file, st_dev_id)
+                    return st_dev_id
```
Comment (YongxueHong) [inline qemu/tests/virtio_fs_share_data.py]: Suggest this part move top.

File: qemu/tests/cfg/virtio_fs_share_data.cfg [hunk 2]
```diff
@@ -56,6 +56,20 @@
             only Windows
             only default.default.with_cache.auto.default
             viofs_log_file_cmd = 'dir ${viofs_log_file}'
+        - with_stdev_check:
+            no Windows
+            only default.default.with_cache.auto.default
+            fs_binary_extra_options += " -o announce_submounts"
```
[the remark this answers is not in the document]
Comment (xiagao) [reply qemu/tests/cfg/virtio_fs_share_data.cfg]: Update thanks.

File: qemu/tests/virtio_fs_share_data.py (hunk 1 above)
Comment (xiagao) [reply qemu/tests/virtio_fs_share_data.py]: update.

**Comment by xiagao** [comment]
(1/1) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.virtio_fs_share_data.with_cache.auto.with_stdev_check.q35: PASS (110.74 s)
@YongxueHong  Could you have a check? thanks

**Comment by xiagao** [comment]
@zhenyzha  Could you help test ppc part? Thanks.

File: qemu/tests/virtio_fs_share_data.py [hunk 3]
```diff
@@ -104,24 +112,36 @@ def get_viofs_exe(session):
...
+            nfs_params = params.object_params(fs)
+
+            params["export_dir"] = nfs_params.get("export_dir")
+            params["nfs_mount_src"] = nfs_params.get("nfs_mount_src")
+            params["nfs_mount_dir"] = nfs_params.get("fs_source_dir")
+            if cmd_get_stdev:
+                fs_source_dir = nfs_params.get("fs_source_dir")
+                params["nfs_mount_dir"] = os.path.join(fs_source_dir, nfs_mount_dst_name)
+            nfs_local = nfs.Nfs(params)
+            nfs_local.setup()
```
Comment (YongxueHong) [inline qemu/tests/virtio_fs_share_data.py:144]: The original codes is `env_process.preprocess(test, params, env) ` if `cmd_xfstest` is True,so we better to keep it.
I suggest that:
```

if cmd_xfstest:
    ...
if setup_local_nfs:
    ...
if cmd_xfstest or setup_local_nfs:
    params["start_vm"] = "yes"
    env_process.preprocess(test, params, env)    

```
This looks like more readable.

**Review by YongxueHong — APPROVED** [review]
LGTM.

**Comment by xiagao** [comment]
(1/4) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.virtio_fs_share_data.with_cache.auto.q35: PASS (75.72 s)
 (2/4) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.virtio_fs_share_data.with_cache.auto.with_stdev_check.q35: PASS (91.00 s)
 (3/4) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.virtio_fs_share_data.run_stress.with_xfstest.with_local_source.with_cache.auto.q35: PASS (202.69 s)
 (4/4) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.virtio_fs_share_data.run_stress.with_xfstest.with_nfs_source.with_cache.auto.q35: PASS (166.09 s)
update @YongxueHong

**Comment by YongxueHong** [comment]
Hi, @zhenyzha 
Could you help me to check whether review it on ppc side?
Thanks.

**Comment by zhenyzha** [comment]
> Hi, @zhenyzha
> Could you help me to check whether review it on ppc side?
> Thanks.

Got it, building env, update results today.

**Comment by zhenyzha** [comment]
```
 (1/1) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.virtio_fs_share_data.with_cache.auto.with_stdev_check.arm64-pci: PASS (285.27 s)
 (1/1) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.aarch64.io-github-autotest-qemu.virtio_fs_share_data.with_cache.auto.with_stdev_check.arm64-pci: PASS (220.50 s)
```

And xfstest.with_nfs_source can be running, 
LGTM.Acked-by: Zhenyu Zhang <zhenyzha@redhat.com>

**Review by zhenyzha — APPROVED** [review]

**Pull request merged**